File: e2e_tests/python/server_clients/tool.py
from typing import Any


class Tool:
    """Represents a tool with its properties and formatting."""

    def __init__(
        self, name: str, description: str, input_schema: dict[str, Any]
    ) -> None:
        self.name: str = name
        self.description: str = description
        self.input_schema: dict[str, Any] = input_schema

    def format_for_llm(self) -> dict:
        """Format tool information for the Bedrock Converse API.

        Returns:
            A tool specification dictionary.
        """
        return {
            "toolSpec": {
                "name": self.name,
                "description": self.description,
                "inputSchema": {
                    "json": self.input_schema,
                },
            }
        }

File: examples/chatbot-typescript/src/server_clients/tool.ts
/**
 * Represents a tool with its properties and formatting.
 */
export class Tool {
  name: string;
  description: string;
  inputSchema: Record<string, any>;

  constructor(name: string, description: string, inputSchema: Record<string, any>) {
    this.name = name;
    this.description = description;
    this.inputSchema = inputSchema;
  }

  /**
   * Format tool information for the Bedrock Converse API.
   * @returns A tool specification dictionary.
   */
  formatForLLM(): Record<string, any> {
    return {
      toolSpec: {
        name: this.name,
        description: this.description,
        inputSchema: {
          json: this.inputSchema,
        },
      },
    };
  }
}
